### PR TITLE
Adding Casbin for User Roles

### LINKED ISSUE #100: User roles and restrictions
### User Story
As an administrator
I'd like to be sure that certain tasks in the system cannot be performed by non-admins
So that I can be sure of the security of the system

### Acceptance Criteria
- [x] `userRole` endpoints re-activated
- [x] `admin` role created by default in-code (not seeder)
- [x] `admin` user created by code (_optional_)
- [ ] PR merged to `master`
- [x] Following actions are limited to ONLY admin users: 
          - Create, edit and delete user accounts
          - Create, edit and delete entities
          - Create, edit and delete questionnaires
          - Update and delete check-ins
- [x] Admin role should be able to do anything, not restricted.

### Notes
For now, we'll only create a default `admin` role. Any other roles and functionality will be a future consideration.

### LINKED ISSUE #101: Login restrictions
### User Story
As an end user
I'd like to have my login locked after a certain number of failed login attempts
So that I can be sure brute force methods of login won't work.

### Acceptance Criteria
- [x] A method to restrict logins is researched and implemented.
- [ ] PR merged to `master`

### PR DESCRIPTION
**Pull request by revjtanton**
We are using Casbin to administer user roles and privvies. This work
will add Casbin integration as well as Casbin/Sequelize integration.
This includes model, migrate, and seeder.

Additionally we are adding brute force protections with this PR.
resolves #100 
resolves #101 

## Todos

- [x] Tests
- [x] Documentation

## Deploy Notes

This will require a re-creation and re-seeding of the db.

## Impacted Areas in Application

List general components of the application that this PR will affect:

- roles
- security

### CODE REVIEW

**Comment by revjtanton** [comment]
@allcontributors add @blakenan-bellese for ideas

File: sequelize/data/user-role.json [hunk 1]
```diff
@@ -1,6 +1,73 @@
 [
     {
         "role": "admin",
```
Comment (c-w-allen) [inline sequelize/data/user-role.json:3]: Is there an easier way to just let an admin do everything here? Just curious

File: src/index.js [hunk 2]
```diff
@@ -39,14 +43,14 @@ app.use(async (req, res, next) => {
 });
 
 // Helper endpoints
-app.get('/', (req, res) => {
-	res.redirect('/api-docs');
+app.get('/', apiLimiter, (req, res) => {
```
Comment (c-w-allen) [inline src/index.js:46]: If you're using the rate limiter everywhere, I think you can do `app.use(apiLimiter)` on its own?

File: sequelize/data/user-role.json (hunk 1 above)
Comment (revjtanton) [reply sequelize/data/user-role.json:3]: Unfortunately from what I've seen with Casbin you cannot wildcard the method, it has to be explicit.

File: src/index.js (hunk 2 above)
Comment (revjtanton) [reply src/index.js:46]: The `apiLimiter` is set to one thing for everything, and something else for login. As this matures different endpoints may require different rules depending on public/private/API-access rules we may want to work out.

Comment (revjtanton) [reply src/index.js:46]: That said I will look into how middleware is overwritten. That is probably a cleaner solution.

**Pull request merged**